yarn-project/types/src/l1_to_l2_message.ts

```diff
@@ -34,7 +46,16 @@ export class L1ToL2Message {
      * The fee for the message.
      */
     public readonly fee: number,
-  ) {}
+    /**
+     * The entry key for the message - optional.
+     * If not provided, it will be calculated by hashing the message (similar to the Inbox contract)
+     */
+    public readonly entryKey?: Fr,
+  ) {
+    if (!entryKey) {
+      this.entryKey = this.hash();
```

Comment:
speaking of this i dont think the hash on this class matches the sol just yet (my bad)